core/src/test/scala/com/banno/kafka/ConsumerAndProducerApiSpec.scala

```diff
@@ -97,7 +97,7 @@ class ConsumerAndProducerApiSpec
             _ <- c.assign(topic, Map.empty[TopicPartition, Long])
             f <- Concurrent[IO].start(c.poll(1 second))
             e <- Temporal[IO].sleep(10.millis) *> c.close.attempt
-            _ <- f.join
+            _ <- f.joinWithNever
```

Comment:
`never` could make it hang right?

Reply:
Yeah, that's why the commit message is `this will likely make things worse, but who knows`. 

Total shot in the dark, based on the fact that this previously used CE2's `join` without issue, and that had the same semantics as CE3's `joinWithNever`.

Reply:
LOL - and it's now green for the first time today. Going to give it a couple retries.

Reply:
I can't really come up with an explanation for why reverting to "hang when canceled" semantics would have any positive impact. 

It's possible that the switch to CE3 `join` brought to the fore some latent race condition, and `joinWithNever` is merely hiding it again, which would be a bad thing.